kolibri/plugins/user_profile/assets/src/machines/changeFacilityMachine.js

```diff
@@ -81,9 +81,16 @@ export const changeFacilityMachine = createMachine({
   context: {
     role: 'learner',
     username: '',
-    targetAccount: { username: '', id: '' },
+    targetAccount: {
+      fullName: '',
```

Comment:
Where should I get it and what do you mean exactly by the "final merge"? `CreateAccount` page only collects information from users who wish to create a completely new account - their username, full name, and password. That account will be created by the backend later on. I assume we don't generate ids for new users on front-end?

 It's different from merging accounts - in that case I assume we will have `id` available.

I guess I could add `id` back with a comment that it can be empty in some cases but maybe it'd make more sense to have two structures in the state machine's context to differentiate these scenarios, `newAccount` and `mergeTargetAccount`?

Reply:
It's one of the closest issues in this epic, the PR that implemented it is this one: https://github.com/learningequality/kolibri/pull/9454 , this https://github.com/learningequality/kolibri/blob/develop/kolibri/plugins/user_profile/tasks.py#L62-L78 documents the task that will do the final merge.

Reply:
I see, from the code it seems you're using "merge" term for both scenarios - creating a new account as well as merging with an existing one. I think I get that now, but still don't understand where you suggest I should get an id when someone's creating a new account. Could you please explain what `id`'s value should be?

Reply:
nah, forget it. I lost the reference in this task, looking from/to local/remote all the time made me lose the thread. The remote user id will "appear" in the local machine after they sync in the task I reference above. After the remote user is synced to the local machine the task gets the id using the username at https://github.com/learningequality/kolibri/blob/develop/kolibri/plugins/user_profile/tasks.py#L84. So what we really need from the remote facility is what exposed in the validator: its `facility_id`, `username` and `password` (the latest only if the user exists and needs to authenticate).Neither  `fullName` nor `id` are not required.